File: lab_2/generator.java
import java.util.Random;

/**
* This class provides a method to generate a random sequence of bits.
*/
public class Main
{

/**
 * Generates a random sequence of bits.
 *
 * @param rand Creates a new object of class Random.
 * @param randNum The number generated at random.
 * @param binaryNum Defines what value will be output to the console: "1" or "0".
 * @return A string representing the random bit sequence.
 */
  public static void main(String[] args) {

        Random rand = new Random(System.currentTimeMillis());

        for (int i = 0; i < 128; i++) {
            long randNum = rand.nextInt(32767);
            boolean binaryNum = randNum % 2 == 1;
            System.out.print(binaryNum ? "1" : "0");
        }
    }
}

File: lab_2/generator.cpp
#include <iostream>
#include <random>
#include <ctime>

#define MAX_BIT 128

using namespace std;

/**
 * Generates a sequence of 128 random binary numbers.
 *
 * @param [in] rand_num generates a random number in the range from 0 to 32766.
 * @param [in] binary_num forms a binary sequence from the remainder of even and odd values
 * @return The randomly generated binary string.
 */
int generator()
{
    srand(time(0));
    for (int i = 0; i < MAX_BIT; ++i)
    {
        unsigned long long rand_num = rand() % 32767;
        bool binary_num = rand_num % 2;
        cout << binary_num;
    }
    return 0;
}

int main()
{
    cout << generator() << endl;
}